### DISCUSSION TITLE
Consider cutting a 2.0.4 with RA type resolution fix

### DISCUSSION THREAD

**Discussion by adamchalmers** [diesel-rs/diesel#3594] [Ideas]
Hi @weiznich,

Firstly, thank you for your work on https://github.com/diesel-rs/diesel/commit/b024f2066147b87c0c6d51dd150fe9d95e0084da to fix Rust Analyzer. It's hard to work on a large Diesel project without RA, so I'm glad you were able to fix it in Diesel! It's a shame RA doesn't support the 2015-style resolution, but I'm glad Diesel could work around it.

I was going to change my project to use diesel as a git dependency, so that I can pull that commit into my project and get RA working again. But my project uses several other crates that all use diesel 2.0 from crates.io.

In my opinion, a lot of Rust programmers are probably affected by RA-diesel problems, so maybe it's worth cutting a new branch from 2.0.3, cherry-picking the type resolution commit in, and releasing that as diesel 2.0.4?

**Comment by weiznich** [comment]
I'm happy to issue another patch release there if someone backports the relevant PR. If we do another patch release I would like to include the following PR's:

* https://github.com/diesel-rs/diesel/pull/3569
* https://github.com/diesel-rs/diesel/pull/3563
* https://github.com/diesel-rs/diesel/pull/3560
* https://github.com/diesel-rs/diesel/pull/3564
* https://github.com/diesel-rs/diesel/pull/3589

**Comment by adamchalmers** [comment]
OK, let me give that a try.

**Comment by adamchalmers** [comment]
I tried cutting the hotfix branch, but I'm having trouble cherry-picking https://github.com/diesel-rs/diesel/pull/3563 -- my apologies but I'm not familiar enough with the internals to resolve these conflicts.

**Reply by weiznich** [reply]
It might be fine to postpone the syn update to the next feature release. So if that's the only problematic PR, just drop it.

**Comment by adamchalmers** [comment]


**Reply by weiznich** [reply]
Perfect, thanks :+1:
I've opened a PR to the correct branch and will do a release on Monday if no other member from the core team has opinions on this. Could you remove the added file in diesel_derives? That's an artifact from back-porting. It shouldn't be there.

**Comment by adamchalmers** [comment]
Sorry, I deleted my comment because I noticed the same problem. I'm trying to ensure I don't regress on the quote issue.

PR is https://github.com/diesel-rs/diesel/pull/3597
